internal/provisioner/utility_group.go

```diff
@@ -0,0 +1,69 @@
+package provisioner
+
+import (
+	"github.com/mattermost/mattermost-cloud/internal/tools/aws"
+	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
+	"github.com/mattermost/mattermost-cloud/model"
+)
+
+// ClusterUtility is a third party service that runs one per cluster and must be managed at cluster creation/deletion time. It is responsible for management of its own lifetime, given a handle pointing to the correct cluster.
+type ClusterUtility interface {
+	Create(group UtilityGroup) error
+	Upgrade(group UtilityGroup) error
+	Destroy(group UtilityGroup) error
+}
+
+// UtilityGroup holds the metadata needed to manage a specific UtilityGroup, and therefore uniquely identifies one, and can be thought of as a handle to the real group of utilities running inside of the cluster
+type UtilityGroup struct {
+	cluster     *model.Cluster
+	provisioner *KopsProvisioner
+	awsClient   aws.AWS
+	kops        *kops.Cmd
+}
+
+// NOTE: if you add a utility to this list, the ORDER MATTERS.
+// The services will be provisioned IN ORDER and if something is provisioned before its dependencies it WILL FAIL
+var clusterUtilities []ClusterUtility = []ClusterUtility{
+	&nginx{}, &prometheus{}, &fluentbit{},
+}
+
+// CreateUtilityGroup creates and starts all of the third party services needed to run a cluster in parallel.
+func CreateUtilityGroup(group UtilityGroup) error {
+	err := installHelm(group.kops, group.provisioner.logger)
+	if err != nil {
+		return err
+	}
+
+	for _, utility := range clusterUtilities {
+		err := utility.Create(group)
+		if err != nil {
+			return nil
```

Comment:
I like your optimistic approach to not letting errors get in the way of progress, but we should probably return the error here. :)

Reply:
"optimistic" haha, surely I mistyped here..